src/hotspot/share/opto/ifg.cpp

```diff
@@ -409,7 +409,9 @@ uint PhaseChaitin::count_int_pressure(IndexSet* liveout) {
     LRG& lrg = lrgs(lidx);
     if (lrg.mask_is_nonempty_and_up() &&
         !lrg.is_float_or_vector() &&
-        lrg.mask().overlap(*Matcher::idealreg2regmask[Op_RegI])) {
+        (lrg.mask().overlap(*Matcher::idealreg2regmask[Op_RegI]) ||
```

Comment:
Cleanup suggestion: the check grows in size and it is repeated 3 times. Please, extract it into a helper method.